### PR TITLE
git hack: add flag to prompt for parent branch

### LINKED ISSUE #1141: hackable perennial branches
On my current project, we have a release branch that is created at the beginning of each week off develop and everything on it goes to production at the end of the week. Any feature that can be released by the end of the week is merged into it. Otherwise, it should be merged into develop and will be released at the end of the next week (merging down from the release branch to develop is automated). 

Based on this, I have the release branch as a perennial branch and want to be able to more easily create new feature branches off it. Currently, I have to check it out and then run `git append`.

I could create a git alias that does this for me but wanted to bring up the idea of some config that make a perennial branch "hackable". Then when running git hack, it prompts the user to pick between the main development branch and the "hackable" perennial branches for what parent to use. There would be no prompt if the no perennial branches are "hackable".

@kevgo thoughts?

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #1141

### CODE REVIEW

File: src/cmd/hack.go [hunk 1]
```diff
@@ -40,9 +43,18 @@ $ git town new-branch-push-flag false`,
 	},
 }
 
+func getParentBranch(targetBranch string) string {
+	if promptForParent {
+		parentBranch := prompt.AskForBranchParent(targetBranch, git.GetMainBranch())
+		prompt.EnsureKnowsParentBranches([]string{parentBranch})
```
Comment (kevgo) [inline src/cmd/hack.go]: Good idea to check for that here

Comment (charlierudolph) [reply src/cmd/hack.go]: Todo add test that requires this line

Comment (charlierudolph) [reply src/cmd/hack.go]: Test added

**Comment by charlierudolph** [comment]
@kevgo this is ready for another review

**Pull request merged**